states/probe.go

```diff
@@ -120,17 +129,22 @@ func getProbeQueryCmd(cli kv.MetaKV, basePath string) *cobra.Command {
 						cancel()
 						if err != nil {
 							fmt.Printf("Shard %s Leader[%d] failed to search with eventually consistency level, err: %s\n", shard.GetChannelName(), nodeID, err.Error())
+							errCount++
 							continue
 						}
 						if resp.GetStatus().GetErrorCode() != commonpbv2.ErrorCode_Success {
 							fmt.Printf("Shard %s Leader[%d] failed to search,error code: %s reason:%s\n", shard.GetChannelName(), nodeID, resp.GetStatus().GetErrorCode().String(), resp.GetStatus().GetReason())
+							errCount++
 							continue
 						}
 						fmt.Printf("Shard %s leader[%d] probe with search success.\n", shard.GetChannelName(), nodeID)
 					}
 				}
 			}
-
+			if errCount != 0 {
+				fmt.Printf("probe failed, hit %d errors", errCount)
+				os.Exit(-1)
+			}
```

Comment:
This may cause birdwatcher quit in normal mode quit
maybe we shall add a paramter for error handling here?